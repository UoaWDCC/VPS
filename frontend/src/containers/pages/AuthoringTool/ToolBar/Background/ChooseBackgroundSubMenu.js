import React from "react";
import { MenuItem } from "@material-ui/core";
import CloudQueueIcon from "@material-ui/icons/CloudQueue";
import styles from "../../../../../styling/ToolBar.module.scss";
import ChooseBackgroundModal from "./ChooseBackgroundModal";
import useChooseBackgroundModal from "./useChooseBackgroundModal";

function ChooseBackgroundSubMenu() {
  const { isShowing, hide, show } = useChooseBackgroundModal();

  return (
    <MenuItem className={styles.menuItem} onClick={show}>
      <ChooseBackgroundModal isShowing={isShowing} hide={hide} />
      <CloudQueueIcon fontSize="medium" />
      &nbsp;&nbsp;Choose background
    </MenuItem>
  );
}

export default ChooseBackgroundSubMenu;
