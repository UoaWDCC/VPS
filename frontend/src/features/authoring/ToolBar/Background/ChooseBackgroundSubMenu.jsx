import { MenuItem } from "@material-ui/core";
import CloudQueueIcon from "@material-ui/icons/CloudQueue";
import styles from "../ToolBar.module.scss";
import ChooseBackgroundModal from "./ChooseBackgroundModal";
import useChooseBackgroundModal from "./useChooseBackgroundModal";

/**
 * This component is the submenu for for background component
 * Beaware that the modal needs to be rendered outside of the MenuItem: https://stackoverflow.com/a/41991884
 * @component
 */
function ChooseBackgroundSubMenu() {
  const { isShowing, hide, show } = useChooseBackgroundModal();
  return (
    <>
      <MenuItem className={styles.menuItem} onClick={show}>
        <CloudQueueIcon fontSize="medium" />
        &nbsp;&nbsp;Choose Image
      </MenuItem>
      <ChooseBackgroundModal isShowing={isShowing} hide={hide} />
    </>
  );
}

export default ChooseBackgroundSubMenu;
