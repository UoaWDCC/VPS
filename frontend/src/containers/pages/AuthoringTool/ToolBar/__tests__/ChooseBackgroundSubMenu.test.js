import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import ChooseBackgroundSubMenu from "../Background/ChooseBackgroundSubMenu";

test("Scenario Selection page snapshot test", () => {
  const component = renderer.create(
    <BrowserRouter>
      <ChooseBackgroundSubMenu />
    </BrowserRouter>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
