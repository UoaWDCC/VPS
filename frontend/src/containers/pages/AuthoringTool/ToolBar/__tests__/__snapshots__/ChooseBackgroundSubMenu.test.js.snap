// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseBackgroundSubMenu component snapshot test 1`] = `
Array [
  <li
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root menuItem MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="menuitem"
    tabIndex={-1}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM19 18H6c-2.21 0-4-1.79-4-4s1.79-4 4-4h.71C7.37 7.69 9.48 6 12 6c3.04 0 5.5 2.46 5.5 5.5v.5H19c1.66 0 3 1.34 3 3s-1.34 3-3 3z"
      />
    </svg>
      Choose background
  </li>,
  <div />,
]
`;
